app/src/main/java/zapsolutions/zap/connection/LndConnection.java

```diff
@@ -59,7 +60,7 @@ private void readSavedConnectionInfo() {
         mConnectionInfo = connectionInfo.split(";");
         ZapLog.debug(LOG_TAG, connectionInfo);
 
-        if (mConnectionInfo[2].equals("null")) {
+        if (mConnectionInfo[2].equals(BTCPayConfig.NO_CERT)) {
             // BTC PAY (no cert, hex encoded macaroon)
             mMacaroon = new MacaroonCallCredential(mConnectionInfo[3]);
         } else {
```

Comment:
In the else part the macaroon actually has to be created like this:


            // LND Connect
            String macaroonBase64UrlString = mConnectionInfo[3];

            if (mConnectionInfo[2].equals("null")){
                mMacaroon = new MacaroonCallCredential(mConnectionInfo[3]);
            } else {
                byte[] macaroonBytes = BaseEncoding.base64Url().decode(macaroonBase64UrlString);
                String macaroon = BaseEncoding.base16().encode(macaroonBytes);
                mMacaroon = new MacaroonCallCredential(macaroon);
            }

This covers the case when we want to connect to a BtcPay with an lndconnect string.
In the future we will have to make sure that all our parsers already encode the macaroon in the same manner, so on the device the connection information is saved in the same format, no matter where it came from.
But this would require to reconnect the node after an update within the app.
One of the next things will probably Biometrics which will also require a reconnect. That's why I would wait with this refactor, so we need to do this only once.

Reply:
Yes I agree, we should try to move the persisted config to a general format.
Biometrics are optional, so they would only require a reconnect for people that will use them, right?

What we could do is to support a legacy version for a few releases and with time stop supporting it, e.g. when the user base drops to a certain percentage of that version.